File: models/Event.js
const mongoose = require("mongoose");
// const formatISO = require("date-fns/formatISO");
// const currentDate = formatISO(new Date());

const EventSchema = new mongoose.Schema(
    {
        title: {
            type: String,
            require: true,
        },
        category: {
            type: String,
            require: true,
        },
        address: {
            type: String,
            require: true,
        },
        lat: {
            type: Number,
            require: true,
        },
        long: {
            type: Number,
            require: true,
        },
        startTime: {
            type: String,
            // require: true,
        },
        endTime: {
            type: String,
            // require: true,
        },
        createdBy: {
            type: String,
            description: ""
        },
        description: {
            type: String,
        },
        likes: {
            type: Number,
            require: true,
            min: 0,
        },
        capacity: {
            type: Number,
            require: true,
            // min: 1,
            // max: 20,
        }
    },
    // {timestamps: true}
);

module.exports = mongoose.model("Event", EventSchema);

File: Event.swift
//
//  Event.swift
//  eventU
//
//  Created by Alexandra Brown on 6/28/21.
//

import Foundation
import SwiftUI

struct Event: Hashable, Codable, Identifiable {
    var id: Int
    var objId: String
    var title: String
    var category: Category
    var name: String
    var street: String
    var city: String
    var state: String
    var lat: Double
    var long: Double
    var startTime: Date
    var endTime: Date
    var description: String
    var likes: Int
    var createdBy: String
    var capacity: String
    var deleted: Int
    
    static let `default` = Event(id: 0, objId: "60e05d9b93ea063127da6ddc", title: "concert", category: Category.music, name: "UCF", street: "4405 central florida blvd", city: "orlando", state: "florida", lat: 28.58609, long: -81.194122, startTime: Date(), endTime: Date(), description: "a concert", likes: 4, createdBy: "abrown", capacity: "20", deleted: 0)
    
}

